### PR TITLE
Offer Open as Folder when no repos are selected in folder import

### LINKED ISSUE #6647: Add Project: "Import repositories from folder" dead-ends at 0 selected; offer the existing "Open as Folder" action
## Summary
When Add Project routes you into "Import repositories from folder" (a folder that is not itself a Git repo but contains nested repos), deselecting all repositories disables **both** action buttons with no explanation and no way forward. The only exits are the small Back arrow and the X. It reads as broken.

The cleanest resolution is also the most useful one: when nothing is selected, offer the **Open as Folder** action that already exists elsewhere in the app, so the user can add the parent as a plain (non-Git) folder workspace instead of being stuck.

## Screenshot
Add Project import dialog, "0 of 46 selected", both "No, import separately" and "Yes, import as group" disabled:

Orca Add Project import dialog: 0 of 46 selected, both 'No, import separately' and 'Yes, import as group' disabled

## Repro
1. Add Project -> Browse -> pick a folder that contains nested repos. The import dialog opens.
2. Leave it at, or set it to, "0 of N selected."
3. Both "No, import separately" and "Yes, import as group" are greyed out.
4. No tooltip, no helper text, no enabled action. Nothing explains what to do.

## Expected
The dialog should never present a stuck state with zero guidance. At 0 selected it should offer a way out, ideally **Open as Folder** so the visit is not wasted.

## Actual
Two disabled buttons, no message, no tooltip, no obvious exit. Looks like a bug even though the disable is intentional.

## Root cause (refs `main`)
Both buttons share one gate, and nothing is rendered for the empty-selection state. `src/renderer/src/components/sidebar/AddRepoNestedImportStep.tsx:143` and `:154`:
```ts
disabled={isAdding || scanInProgress || selectedPaths.size === 0}
```
There is no `selectedPaths.size === 0` branch that surfaces guidance or an alternative action.

## Proposed resolution

**Primary: surface "Open as Folder" here.** This is not new functionality. The capability already exists:
- the `NonGitFolderDialog` "Open as Folder" flow and the `addNonGitFolder` store action,
- `kind: 'folder'` projects (`src/main/ipc/repos.ts` `addLocalRepoFromPath(..., kind: 'git' | 'folder')`),
- and the local flow already adds non-Git folders this way when no nested repos are present (`src/renderer/src/components/sidebar/useAddRepoLocalFolderFlow.ts`, the `non_git_folder` branch).

The only gap is that the moment a nested repo is found, the user is routed into this import dialog, which has no path back to "just open the folder." Wiring the existing `addNonGitFolder(scan.selectedPath)` action into this dialog (shown/enabled when 0 are selected) closes that gap with code that already exists.

This also mirrors a pattern the team is already shipping: **#6171** adds a third option ("Convert to Git Repository") to the sibling non-Git dialog. "Open as Folder" here is the mirror image of that.

**Fallback (if Open as Folder is out of scope):** at minimum, when `selectedPaths.size === 0`, render a one-line hint or a tooltip on the disabled buttons, e.g. "Select at least one repository to import, or press Cancel," and make Cancel visible alongside the disabled actions. Pure presentation change.

## Related (not duplicates)
- #6171 adds "Convert to Git Repository" to the non-Git dialog (different dialog; sets the precedent referenced above).
- #6362 fixes monorepo subfolder import identity (unrelated).

## Environment
- Orca.app bundle **1.4.103** (Homebrew cask `stablyai/orca`); CLI shipped in-bundle.
- macOS, Apple Silicon.
- Code refs: `stablyai/orca@main` as of 2026-06-28.

### PR DESCRIPTION
**Pull request by brennanb2025**
## What changes

In Add Project → "Import repositories from folder", deselecting **every** repository disabled both action buttons ("No, import separately" and "Yes, import as group") with no enabled action and no explanation — a dead-end. This adds, **only at zero selected**, an enabled secondary **"Open as Folder"** button plus a one-line hint, so the user can add the parent as a plain non-Git folder workspace instead of being stuck.

## What does not change

The one-or-more-selected path is untouched: when at least one repository is selected, the step renders exactly as before (no extra button, no hint, both import buttons enabled). The fix does not change scanning, import, or grouping behavior.

## How it works

The "Open as Folder" action routes to the existing `confirm-non-git-folder` dialog (`NonGitFolderDialog`), reusing the shipped folder-add flow rather than adding a new one. That dialog already handles all three contexts — local, runtime, and SSH — and shows the capability-downgrade explanation ("Git-based features won't be available"). The handler captures the scan context (path + SSH/runtime identity) into locals before resetting the Add Project dialog's local state, then opens the confirm modal as the last modal store write so it survives the reset.

A telemetry enum member `open_as_folder` was added to the existing `add_repo_nested_import_action` event (alongside `back` / `import_group` / `import_separate`) so the escape-hatch usage is measurable. No new event type, no new fields, no path or repo-name data.

## Design approach

Surface the existing capability at the empty-selection state rather than re-implementing folder-add inside the import step. Scope is intentionally narrow: empty-selection only, mirroring the issue's primary ask. An alternative (show "Open as Folder" persistently regardless of selection) was considered and rejected to keep the common multi-select row focused on the two import CTAs.

## Design-review outcome

Ran Brennan's design-review loop to convergence (clean, no P0/P1/P2). It corrected the modal-handoff ordering to follow the two shipped `confirm-non-git-folder` callers (`closeModal`/reset → `openModal` last) and clarified that the button label reuses the existing translated "Open as Folder" string while the hint is a new key.

## Implementation and deviations

- `AddRepoNestedImportStep.tsx`: at 0 selected, render the hint + a secondary (`outline`) "Open as Folder" button; keep the two import buttons rendered-but-disabled. The button is enabled only when not adding, not scanning, and `scan.selectedPath` is non-empty.
- `AddRepoDialogStepContent.tsx`: pass-through prop.
- `AddRepoDialog.tsx`: wire the handler.
- `use-open-nested-folder-as-folder.ts` (new): the handler (capture-before-reset, telemetry, SSH/runtime/local routing).
- `use-add-repo-dialog-reset.ts` (new): extracted the dialog's reset orchestration — a faithful, behavior-preserving extraction needed to keep `AddRepoDialog.tsx` under the `max-lines` limit instead of disabling the rule.
- `nested-repo-telemetry.ts`: add `open_as_folder` to the action enum.

No deviations from the reviewed design. The reset extraction is the only structural change beyond the doc, made solely to satisfy `max-lines` without a lint disable.

## Completeness verification

A read-only verification pass confirmed every design requirement, edge case, and validation item is implemented (verdict: COMPLETE). The one gap it flagged — a schema-parse test for the `open_as_folder` payload — was then added.

## Headline behavior status

**Implemented.** The dead-end is resolved by a real, production-executing action: at 0 selected the enabled "Open as Folder" button routes through the shipped confirm dialog and adds the folder as a `kind: 'folder'` workspace. Verified end-to-end in Electron (below).

## Perf audit

Perf-neutral. Touched surface is the Add Project dialog only (lazy-mounted, gated on `activeModal === 'add-repo'`) — nothing runs at startup or in any polling/subprocess loop. The change is a conditional render with an O(1) guard plus a `useCallback`-stable handler wired through the stable `openModal` store action; `React.memo` on `AddRepoDialog` is not defeated. No new listeners/timers/subscriptions. No measurement or regression test warranted.

## Code-review loop

One round, two reviewers in parallel (independent perspectives). Both returned clean: Reviewer A found no P0/P1/P2 (only P3 nits); Reviewer B found no issues. Both independently traced the modal handoff and confirmed `resetState()` does not clobber the freshly-opened confirm modal, and that SSH/runtime/local routing matches the existing callers.

## Validation in Electron (merge-confidence)

Launched the dev build from this worktree (CDP), created a non-Git parent folder with 3 nested git repos, and exercised the real flow:

1. Import step shows "Found 3 repositories", "3 of 3 selected", both import buttons enabled.
2. Deselect all → "0 of 3 selected": the hint "Select at least one repository to import, or open the folder as-is." appears, the **enabled** "Open as Folder" button appears, and both import buttons are disabled — **no dead-end**.
3. Click "Open as Folder" → the `confirm-non-git-folder` dialog opens with the correct local folder path and capability-downgrade copy.
4. Confirm → the folder is added as a `kind: 'folder'` workspace, activated, and revealed in the sidebar with a terminal + file tree (nested `api`/`billing`/`web` visible).

Screenshots are attached to the PR conversation.

## Static checks and tests

- `pnpm run lint` (oxlint + switch-exhaustiveness + `verify:localization-catalog` + `verify:localization-coverage`): pass.
- `pnpm run typecheck:web`: pass.
- Targeted vitest (4 files, 30 tests): pass — covers 0-selected enabled+hint+click, ≥1-selected unchanged, scanning-disabled, empty-path-disabled, and the `open_as_folder` telemetry payload through both the builder and `addRepoNestedImportActionSchema`.

## Cross-platform / SSH

No path-string manipulation added (`scan.selectedPath` passed verbatim). SSH routes via `connectionId`, runtime via `runtimeEnvironmentId`, local with neither — identical to the existing `confirm-non-git-folder` callers. macOS/Linux/Windows unaffected.

## Remaining non-actionable notes

- The new hint string is English in the es/ja/ko/zh catalogs (the button label is fully translated everywhere). This matches the repo's established backfill pattern; `verify:localization-catalog` passes with full key parity.
- With three buttons at 0 selected, the row may wrap "Yes, import as group" to a second line in a narrow dialog. It is disabled in that state and the layout is functional; not a regression.

Closes #6647

Made with [Orca](https://github.com/stablyai/orca) 🐋

### CODE REVIEW

File: src/renderer/src/components/sidebar/use-add-repo-dialog-reset.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    resetRemoteState()
+  }, [
+    resetCloneFlow,
+    resetCreateDefaultState,
+    resetCreateState,
+    resetRemoteState,
+    resetServerPathFlow,
+    setAddProjectBusyLabel,
+    setIsAdding
+  ])
```
[the remark this answers is not in the document]
Comment (brennanb2025) [reply src/renderer/src/components/sidebar/use-add-repo-dialog-reset.ts:92]: Good catch — addressed in f7685148c. `resetHostScopedState` now clears the nested import/review state and returns a stranded `nested` step to `add` on a host switch, so a later import / open-as-folder can't combine the prior host's paths with the new host. Added a regression test in `use-add-repo-dialog-reset.test.ts`.

**Comment by nwparker** [comment]
Closing as **superseded** — the linked issue(s) for this PR were fixed by another change that has already been merged.

- Issue #6647 was fixed by #6719 (`Fix empty nested import folder fallback`)

Superseding merge: #6719

Thank you for the contribution! If anything remains unaddressed, please open a new issue or PR against current `main`.
